pkg/master/thirdparty/thirdparty.go

```diff
@@ -84,12 +88,25 @@ func NewThirdPartyResourceServer(genericAPIServer *genericapiserver.GenericAPISe
 	return ret
 }
 
-// thirdPartyEntry combines objects storage and API group into one struct
+type thirdPartyGroup struct {
+	versionedEntry map[string]*thirdPartyEntry
+	group          metav1.APIGroup
+}
+
+// thirdPartyEntry combines objects storage into one struct
 // for easy lookup.
 type thirdPartyEntry struct {
-	// Map from plural resource name to entry
+	// Map from plural resource name to RESTStorage
 	storage map[string]*thirdpartyresourcedataetcd.REST
-	group   metav1.APIGroup
+}
+
+func (g *thirdPartyGroup) versionInstalled(version string) bool {
+	for installedVersion := range g.versionedEntry {
+		if reflect.DeepEqual(version, installedVersion) {
+			return true
+		}
+	}
+	return false
 }
 
 // HasThirdPartyResource returns true if a particular third party resource currently installed.
```

Comment:
if we change `HasThirdPartyResource(rsrc)` to `HasThirdPartyResource(gvr)`, then I would suppose we should make the same change to `InstallThirdPartyResource`, which means we should update the thirdparty controller. wdyt?